source/opt/debug_info_manager.cpp

```diff
@@ -633,16 +708,29 @@ void DebugInfoManager::AnalyzeDebugInst(Instruction* inst) {
     deref_operation_ = inst;
   }
 
+  if (deref_operation_ == nullptr &&
+      inst->GetVulkan100DebugOpcode() ==
+          NonSemanticVulkanDebugInfo100DebugOperation) {
+    const analysis::Constant* operation_const =
+        context()->get_constant_mgr()->GetConstantFromInst(
+            context()->get_def_use_mgr()->GetDef(inst->GetSingleWordOperand(
+                kDebugOperationOperandOperationIndex)));
+
+    if (operation_const->GetU32() != NonSemanticVulkanDebugInfo100Deref) {
```

Comment:
`deref_operation_` must be an instruction whose debug instruction type is the `Deref` operation.
We keep it to reuse it instead of creating multiple `Deref` operations.

Is this if-statement condition correct?
I think it has to be `operation_const->GetU32() == NonSemanticVulkanDebugInfo100Deref`.

Reply:
Good catch! Thanks!